src/dependencies.js

```diff
@@ -84,25 +84,38 @@ const getFileDependencies = async function({ path, packageJson, state }) {
   const dependencies = precinct.paperwork(path, { includeCore: false })
 
   const depsPaths = await Promise.all(
-    dependencies.map(dependency => getImportDependencies({ dependency, basedir, packageJson, state }))
+    dependencies.map(dependency => getImportDependencies({ dependency, basedir, packageJson, state, treeShakeNext }))
   )
   return [].concat(...depsPaths)
 }
 
-const getImportDependencies = function({ dependency, basedir, packageJson, state }) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
-    return getTreeShakedDependencies({ dependency, basedir, packageJson, state })
+const getImportDependencies = function({ dependency, basedir, packageJson, state, treeShakeNext }) {
+  const shouldTreeShakeNext = treeShakeNext || isNextOnNetlify(dependency)
+  if (shouldTreeShake(dependency, shouldTreeShakeNext)) {
+    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakeNext: shouldTreeShakeNext })
   }
 
   return getAllDependencies({ dependency, basedir, state, packageJson })
 }
 
+const isNextOnNetlify = function(dependency) {
+  return dependency === './renderNextPage'
```

Comment:
```suggestion
  return basename(dependency, '.js') === 'renderNextPage'
```

To make it more resilient to changes in `next-on-netlify` related to either the optional file extension, or the directories (but not the filename itself). What do you think?